### ISSUE TITLE
CA2012 yields a false positive when using C# 14's extension block feature

### ISSUE THREAD

**Issue by aradalvand** [dotnet/roslyn-analyzers#7789]
Minimal repro:
```cs
await Test().Foo();
await Test().Bar();

static async ValueTask<int> Test() =>
    123;

public static class Extensions
{
    extension(ValueTask<int> t)
    {
        public async ValueTask<int> Foo() =>
            await t;
    }
    public static async ValueTask<int> Bar(this ValueTask<int> t) =>
        await t;
}
```
Line 1 reports a CA2012 diagnostic, whereas line 2 doesn't, even though line 2 is using an extension method that's identical in signature to the one invoked on line 1. It's just that `Foo` is using C# 14's extensions feature.

<img width="2228" height="777" alt="Image" src="https://github.com/user-attachments/assets/340c9cc1-4f82-4963-bddd-85398fc9240a" />

**Comment by andryushchenko** [comment]
Same here https://github.com/dotnet/roslyn-analyzers/issues/7784
